Migrated CodeGallery C# async example (#1601)
dotnet/samples: async/async-and-await/cs/readme.md

---
languages:
- cs
products:
- dotnet-core
- windows
page_type: sample
name: "Async Sample: Asynchronous Programming with Async and Await in C#"
urlFragment: "async-and-await-cs"
description: "A .NET Core WPF application that contains the example method from Asynchronous Progamming with Async and Await in C#."
---
# Async programming with async and await in C#

> This sample is a WPF application written in C# that contains the example method from [Task asynchronous programming model](https://docs.microsoft.com/dotnet/csharp/programming-guide/concepts/async/task-asynchronous-programming-model). The topic gives an overview of asynchronous programming, including when to use it and how to write an async method. This sample contains an async function that is used as an illustration.

Reviewer: ```suggestion
This sample is a WPF application written in C# that contains the example method from [Task asynchronous programming model](https://docs.microsoft.com/dotnet/csharp/programming-guide/concepts/async/task-asynchronous-programming-model). The article gives an overview of asynchronous programming, including when to use it and how to write an async method. This sample contains an async function that is used as an illustration.
```

[async](https://docs.microsoft.com/dotnet/csharp/language-reference/keywords/async) and [await](https://docs.microsoft.com/dotnet/csharp/language-reference/keywords/await) provide all the advantages of traditional asynchronous programming, but with much less effort from the developer. The compiler does the difficult work that the developer used to do, yet the code retains a logical structure that resembles synchronous code.

The example async function in this sample (named `GetStringAsync`) uses an [HttpClient](https://docs.microsoft.com/dotnet/api/system.net.http.httpclient) method to download the contents of a website.

> The code for the *MainWindow.xaml.cs* file from this sample is included in the topic.

Reviewer: ```suggestion
The code for the *MainWindow.xaml.cs* file from this sample is included in the article.
```

## Sample prerequisites

> This sample is written in Visual Basic and targets .NET Core 3.0 running on Windows. It requires the [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.0).

Reviewer: ```suggestion
This sample is written in C# and targets .NET Core 3.0 running on Windows. It requires the [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.0).
```

## Building the sample

To download and run the sample, follow these steps:

 1. Download and unzip the sample.
    
 2. In Visual Studio (2019 or later):
 

>     1. On the menu bar, choose File, Open, Project/Solution.

Reviewer: ```suggestion
    1. On the menu bar, choose **File** > **Open** > **Project/Solution**.
```

 
    2. Navigate to the folder that holds the unzipped sample code, and open the c# project (.csproj) file.
 

>     3. Choose the F5 key, or Ctrl+F5 keys, to run the project.

Reviewer: Parallel changes can be made to the readme for visual basic.

```suggestion
    3. Choose the <kbd>F5</kbd> key to run with debugging, or <kbd>Ctrl</kbd>+<kbd>F5</kbd> keys to run the project without debugging.
```
